src/fsharp/CompileOps.fs

```diff
@@ -2205,7 +2211,7 @@ type TcConfigBuilder =
       mutable referencedDLLs : AssemblyReference list
       mutable projectReferences : IProjectReference list
       mutable knownUnresolvedReferences : UnresolvedAssemblyReference list
-      optimizeForMemory: bool
+      reduceMemoryUsage: ReduceMemoryFlag
```

Comment:
Here might be a good location to give overview of what this flag implies in the implementation.

Will also help when updating https://fsharp.github.io/ in context of those optimization.

Reply:
Sorry, missed that it is done in the .fsi...

Reply:
Yes, there are docs here https://github.com/Microsoft/visualfsharp/pull/4586/files#diff-3835db36ed701c86797a3c95c330f3daR51 but the Compiler Guide is an important place to update too